canvas.go

```diff
@@ -48,8 +52,32 @@ type line struct {
 	p0, p1 image.Point
 }
 
+var (
+	pathSanitizer = regexp.MustCompile(`\W+`)
+	piSanitizer   = regexp.MustCompile(`<\?xml[^>]*>`)
```

Comment:
Done; though due to issues with maintaining the attributes on the `<svg>` tag, I took a stream-based approach.